### DISCUSSION TITLE
Cbus2zigbee

### DISCUSSION THREAD

**Discussion by jasonhollis** [autoSteve/acMqtt#44] [General]
Steve,

I did some reading on this since my last post and I’m curious as to why you would want to do this with an HA box present and integrated with your c-bus?  I can see that if you have some big nice control panels setup in c-bus that you want to expose new Zigbee lights on?  I’m not deep enough into it to understand what the possibilities are in leveraging the same panels tied to HA or swapping them.  Not having really developed a feel for the performance of the NAC2 yet I would assume that HA running on a PI5 or i5/i7 would run rings around the NAC2 from a performance perspective.  Given the investment that Nabu Casa is making in Zigbee, Thread, and Matter tied with the fact that there are now more than a million homes running HA; I would think you would want to run as much automation as possible in HA not on c-bus but I would love to understand your viewpoints on this question.  

I also picked up the fact that Nabu Casa also working on rolling out their own z-wave HW very shortly as well.  I’ve been working with Z-wave at this point for almost a decade and similar to c-bus it just works.  S2 security is orders of magnitude more secure than Zigby.  It’s sub 1ghz so it’s automatically going to go farther on a given amount of power.  I’m running the 800 series devices and keen to check out long range which will now go a couple of miles and support up to 65,000 devices.  The biggest reason that Zigbee and thread are dangerous is the overlap with 2.4 ghz WIFI.  I don’t think 2.4 ghz WIFI is going to go away any time soon based on it having much greater range and the fact that most of the more crappy IOT/automation devices are still on 2.4 only and usually WIFI4.  

Unless I have missed something Zigbee Chanel 26 is the only channel that does not overlap with 2.4 ghz WIFI in Oz.  Hue for example does not even support switching to 26.  I have read that some devices don’t support 26 but I’ve not encountered one yet.  None the less having only one option of 26 and needing to shove your 2.4 GHz WIFI down to 10 and lower to avoid and overlap just sucks in principle.  I have not gotten as down and dirty on Thread yet but the basic problem is exactly the same because it’s the same spectrum.   When I got to 85 Zigbee lights in my last place I was having all sorts of grief.  I have learned how to fine tune it a bunch since then but if it was all Z-wave none of this would be an issue at all.  I’m glad Nabu Casa is taking up the charge with Z-wave and I know that the founder Paulas was at the latest Z-wave alliance global event.  There is just more kool-aid and marketing behind Zigbee.  From a nerd perspective I think it’s much weaker than Z-wave.  The Z-Wave devices for example sensors are also far more configurable in my experience as well.  Zigbee is a bit more simple to configure and reset but z-wave only takes a few more seconds it worth the time spent.  If you are running your setup properly you shouldn’t be pairing and removing devices very frequently after the initial configuration.  I certainly wish that we could buy down lights with same quality as Hue running on Z-wave.  I’m guessing that the key inhibitor is the cost of the 800 series chipset per globe.  I would happily pay an extra $3-5 per device knowing what I know.

It would be nice to see Z-wave roll up under Matter as means of connectivity in the same way that Thread and WIFI already are and then we are sorted.  The 800 series chips already support all global frequencies so that has already been done.  Z-Wave should be the most widely adopted RF connectivity method as a global standard under Matter.

**Comment by autoSteve** [comment]
Some thoughts.

In my opinion, it is about which controller is treated as authoritative. Or preferable. Or that you're more comfortable with. Or more reliable. Or more integrated. Or because history. Or that can be simply sold with the house without a thought, and without a week long training session for the new owner.

> Authoritative and comfortable

For me, my AC does many things, as almost everything is C-Bus here. I do have the odd Zigbee device here and there. Bedside lights. Electric blanket switches. Turntable. A light strip. But every fixed light in the house is C-Bus, every light switch, the pool filter/heating, the irrigation, sweep fans, roller blinds, the HVAC, PIR sensors, and more. For better or worse.

The Zigbee devices would move with me if I moved house.

Home automation here started here with wanting an "all off" button beside my bed so we didn't need to run around the house like crazy people turning things off at night. And that started many, many years ago, in a time when one could not drive to Harvey Norman and get a smart light. They did not exist. Zigbee? A twinkle in the eye of its creator. Z-wave? Other IOT devices? Nope. Clipsal were well ahead of their time, and it just worked.

> Integrated

There are touch panels on house walls here. These expose AC visualisations, and not Home Assistant, so are 'powered' by the 5500NAC. Again, they're fixed infrastructure, and would be sold with the house. They used to visualise prior generations of Clipsal controllers, which have been upgraded over the years.

I can now also operate any Zigbee device in the house from these, hence via the AC, and not reliant on HA.

> More comfortable

The AC features a powerful LUA programming language, and I can code just about anything I want. So I do.

I do Python for a day job, which HA utilises, so am also comfortable with that, and could build or utilise integrations. But there is the fixed and saleable aspect to all of this.

Where does Home Assistant run? As a virtual machine, on a three-node hyperconverged Proxmox cluster in my shed. And that would depart the house with me.

Sure, HA could run on a little Intel box in a cupboard, or even a Pi, but it doesn't.

> More integrated?

Given the Zigbee devices would depart with me, C-Bus to Zigbee is more science project than fixed and saleable infrastructure like a pendant light or the central heating. So does being more integrated matter? Not really for me. It's just more convenient.

Bottom line is that I just want reliable and fixed infrastructure, where any bells and whistles can be taken away with me and still leave the house functional. All IOT devices could, and would be taken away.

> Fit for purpose

So what does HA do for me? Google voice via Nabu Casa for one. (All the assistants in the house would leave with me if I moved.) Nerdy visualisations of my Tesla batteries and Enphase solar. (A new owner would need to roll their own.) Access to control the house while on the road. (Again, new owner can roll their own if they care.) Tell me how much fuel the BMW has. (Again...) Visualise the CCTV. (Again...) Control the volume of my Yamaha amp from my Mac, power the turntable, and set the right amp input, then get up and drop a Craberries vinyl on the platter. All optional for a home owner.

> The future? Nope.

My entire house electrical system is based on a technical legacy. Would it be done differently today? Probably.

**Reply by jasonhollis** [reply]
Steve,

As usual a very elegant response!  

I have a mate with the two powerwalls and 30 panels on Enphase.  He’s got more money than sense.  I put HA in for him and I’ve got his 227 Hue lights instrumented, etc. etc. etc. I noticed that his net power is massive during the day but his batteries are not charging and then he’s paying for grid at night.  I know enough to know it’s not working like it should be and all of the settings in HA match is App and look correct but he’s batteries never get above reserve or at least for the 1.5 months of data I have.  I was freaking out about it but he said he would get it fixed later and left for China for 2 months.  That burns a whole in my brain.  I reckon he’s spent 100K on the system.  He has a huge place.  It should be running negative and instead he’s probably paying 50-80/day for power when he’s there.  I would be loosing my mind and am just considering it LOL.  Short of that the system looks really cool and the HA integration seems rock solid.  Are you somehow tying any of this back to CBUS through HA?

I guess I should play with Proxmox a bit more.  I have read a bit but at the outset it didn’t seem to do anything I couldn’t do with other tools I already had in place.  I have been considering, at the 10,000’ level, how to build out a rock solid two or three node HA cluster though.  I saw a YouTube video of an Aussie guy who was making me giggle it’s been a while back but he wasn’t even using DNS or anything.  

I used to sell clustering products for management/monitoring etc.  I have build Unix clusters going way back.  It’s been a while but I’m unafraid.  Would you say that Proxmox really facilitates the clustering process?  There are so many ways to attack the problem but if I think about it from a “high end” home owner’s perspective they are probably still not going to want to buy a 50K box to get redundant pathing, PS, multi processors so that you could do something more virtually with Kubernetes etc.  It needs to be something cheaper, simple and elegant.  I tend to be dropping in 2-4 1U network elements anyway, so I’ve been contemplating some relatively inexpensive rack mount boxes just to keep things a bit neater versus NUCs.  The one thing I have not really dug into or mastered is some way to run HA Managed OS inside a much larger more capable system e.g. my 50K Qnap.  I was just running HA itself there for a while in a container but there are some advantages to managed OS.  What HA is not considering is why force me to run this on a NUC or something when I already have ridiculous amounts of resources highly available in a much larger system. It’s quite possible that I can simply and I’ve missed something.  I’ve not internalised the complete docker structure of HA to the point that I know it in my head without being able to look something up.  I long ago figured out how to use PI’s on WIFI to drop Z-wave and Zigbee wherever I wanted to have a controller/co-ordinator or an extra one.  

I think two or three 1U Dell servers for example is still extreme overkill.  In the testing I’ve done thus far I reckon a PI5 with NVME and 8 gigs of memory short of Frigate work is already a rocket ship.  Anecdotally it does not feel a ms slower than my i5 with 60 gigs of memory on NVME but I guess the cluster design should be flexible enough to anticipate Frigate/CCTV type stuff in addition to HA.

Your thoughts would be appreciated!

**Comment by autoSteve** [comment]
> Proxmox

It's cool. And free using the community edition. NVMe Ceph storage here in a three-way replica across the nodes, with redundant 10Gb interconnects. It does way more than HA. Web, DNS, reverse proxy, email, Ansible, Puppet, local Gitea, home lab, etc, etc. Clustering is fail-over, with live migration if I need to patch, so hypervisor reboots do not disrupt anything.

In addition to clustering, there is also BGP in the network, so three DNS VMs, three reverse proxy, and seamless fail-over for those in the network layer on UDM pro should a VM hang/reboot. I love it.

It has not failed in any way over seven years so far.

Just to run HA? I'd use a single NUC running Proxmox, and HA as a VM. No cluster. Simple stuff just works. Complex requires care and tending. A single node would give snapshot backup, and also the foundation for backup to a NAS or cloud storage. If I sold, my twin NAS would come too of course, so I wouldn't run HA as a VM on those. The NUC could be left behind and not backed up unless the new owner cared. I would not go anywhere near a Pi, as way too unreliable.

Setting up a hyper-converged Proxmox on NUCs would theoretically be possible, but storage write speed would be limited. And you'd need three minimum. I would not bother with baby compute, unless science project, and not aiming for rock-solid like my Supermicro set up in the shed rack.

> Tying the Tesla/Enphase back to C-Bus?

Absolutely. Automations in HA feed MQTT. acMqtt pulls the sensor values in to the AC objects, then things like state of charge are visualised on wall panel idle screens.

> Your mate with solar/battery

I struggle to fill two Powerwalls anywhere near full with 30 panels on crappy days like today. More like to ~25%. (Charge is 19% right now.) Sunny? That results in both grid feed-in and two days of no power use should day two be crappy. If there's a week of rain? Forget it. What I do then is charge the PWs from grid during the day on off-peak up to a charge that will see me over the peak period. So I never pay more than $0.23-ish per kWh (including charge/discharge losses). It's a fun game that I plan to automate at some stage.

$48k investment here, with modelling suggesting a ten year payback. Reality will be another thing. Because Melbourne.

**Comment by autoSteve** [comment]
And as I hit send the sun came out, and charging is peaking at 10kW _and_ powering the house. Lols.

**Comment by jasonhollis** [comment]
I took a look at a similar system near you, and his **NET** consumption got as low as about -8.5 KW yesterday, which is not bad considering it's a 7-bedroom place with 3 floors full of gadgets.
